rllib/policy/eager_tf_policy.py

```diff
@@ -724,52 +661,130 @@ def set_state(self, state):
             # Then the Policy's (NN) weights.
             super().set_state(state)
 
+        @override(Policy)
+        def export_checkpoint(self, export_dir):
+            raise NotImplementedError  # TODO: implement this
+
+        @override(Policy)
+        def export_model(self, export_dir):
+            raise NotImplementedError  # TODO: implement this
+
         def variables(self):
             """Return the list of all savable variables for this policy."""
             if isinstance(self.model, tf.keras.Model):
                 return self.model.variables
             else:
                 return self.model.variables()
 
-        @override(Policy)
-        def is_recurrent(self):
-            return self._is_recurrent
+        def loss_initialized(self):
+            return self._loss_initialized
 
-        @override(Policy)
-        def num_state_tensors(self):
-            return len(self._state_inputs)
+        @with_lock
+        def _compute_actions_helper(self, input_dict, state_batches, episodes,
+                                    explore, timestep):
+            # Increase the tracing counter to make sure we don't re-trace too
+            # often.
+            self.re_trace_counter += 1
 
-        @override(Policy)
-        def get_initial_state(self):
-            if hasattr(self, "model"):
-                return self.model.get_initial_state()
-            return []
+            # Calculate RNN sequence lengths.
+            batch_size = tree.flatten(input_dict[SampleBatch.OBS])[0].shape[0]
+            seq_lens = tf.ones(batch_size, dtype=tf.int32) if state_batches \
+                else None
 
-        def get_session(self):
-            return None  # None implies eager
+            # Add default and custom fetches.
+            extra_fetches = {}
 
-        def get_placeholder(self, ph):
-            raise ValueError(
-                "get_placeholder() is not allowed in eager mode. Try using "
-                "rllib.utils.tf_utils.make_tf_callable() to write "
-                "functions that work in both graph and eager mode.")
+            # Use Exploration object.
+            with tf.variable_creator_scope(_disallow_var_creation):
+                if action_sampler_fn:
+                    dist_inputs = None
+                    state_out = []
+                    actions, logp = action_sampler_fn(
+                        self,
+                        self.model,
+                        input_dict[SampleBatch.CUR_OBS],
+                        explore=explore,
+                        timestep=timestep,
+                        episodes=episodes)
+                else:
+                    if action_distribution_fn:
 
-        def loss_initialized(self):
-            return self._loss_initialized
+                        # Try new action_distribution_fn signature, supporting
+                        # state_batches and seq_lens.
+                        try:
```

Comment:
try ... except and relying on an arbitrary error message string always gives me bad feelings.
can we remove this now?? how long has it been?? :)

Reply:
Never know, which users are still using the old Policy building APIs (`action_distribution_fn`) and get very upset if we just discontinue support for these. :)

Reply:
It hasn't been too long. Less than a year.